providers/unleash/src/test/java/dev/openfeature/contrib/providers/unleash/UnleashProviderTest.java

```diff
@@ -0,0 +1,272 @@
+        UnleashContext unleashContext = UnleashContext.builder().userId("111").build();
+        EvaluationContext evaluationContext = ContextTransformer.transform(unleashContext);
+        assertEquals(true, unleashProvider.getBooleanEvaluation(flagName, false, evaluationContext).getValue());
+        assertEquals(true, client.getBooleanValue(flagName, false, evaluationContext));
+        unleashContext = UnleashContext.builder().userId("2").build();
+        evaluationContext = ContextTransformer.transform(unleashContext);
+        assertEquals(false, unleashProvider.getBooleanEvaluation(flagName, false, evaluationContext).getValue());
+        assertEquals(false, client.getBooleanValue(flagName, false, evaluationContext));
+    }
+
+    @Test
+    void typeMismatch() {
+        assertThrows(TypeMismatchError.class, () -> {
+            unleashProvider.getIntegerEvaluation("test", 1, new ImmutableContext());
+        });
+        assertThrows(TypeMismatchError.class, () -> {
+            unleashProvider.getDoubleEvaluation("test", 1.0, new ImmutableContext());
+        });
+        assertThrows(TypeMismatchError.class, () -> {
+            unleashProvider.getObjectEvaluation("test", new Value(), new ImmutableContext());
+        });
+    }
+
+    @SneakyThrows
+    @Test
+    void shouldThrowIfNotInitialized() {
+        UnleashProvider asyncInitUnleashProvider = buildUnleashProvider(false, "http://fakeAPI", "{}", new TestSubscriber());
+        assertEquals(ProviderState.NOT_READY, asyncInitUnleashProvider.getState());
+
+        // ErrorCode.PROVIDER_NOT_READY should be returned when evaluated via the client
+        assertThrows(ProviderNotReadyError.class, ()-> asyncInitUnleashProvider.getBooleanEvaluation("fail_not_initialized", false, new ImmutableContext()));
+        assertThrows(ProviderNotReadyError.class, ()-> asyncInitUnleashProvider.getStringEvaluation("fail_not_initialized", "", new ImmutableContext()));
+
+        asyncInitUnleashProvider.initialize(null);
+        assertThrows(GeneralError.class, ()-> asyncInitUnleashProvider.initialize(null));
+
+        asyncInitUnleashProvider.shutdown();
+    }
+
+    @SneakyThrows
+    @Test
+    void shouldThrowIfErrorEvent() {
+        UnleashProvider asyncInitUnleashProvider = buildUnleashProvider(false, "http://fakeAPI", "{}", null);
+        asyncInitUnleashProvider.initialize(new ImmutableContext());
+
+        asyncInitUnleashProvider.emitProviderError(ProviderEventDetails.builder().build());
+
+        // ErrorCode.PROVIDER_NOT_READY should be returned when evaluated via the client
+        assertThrows(GeneralError.class, ()-> asyncInitUnleashProvider.getBooleanEvaluation("fail", false, new ImmutableContext()));
+        assertThrows(GeneralError.class, ()-> asyncInitUnleashProvider.getStringEvaluation("fail", "", new ImmutableContext()));
+
+        asyncInitUnleashProvider.shutdown();
+    }
+
+    @SneakyThrows
+    @Test
+    void contextTransformTest() {
```

Comment:
I am not sure what this actually tests, as one of the method being tested is only used by this test.

Reply:
Looks like this test is testing a "real" method used at runtime now.